### PR TITLE
Make gRPC metadata available to AttributeExtractors

### PR DESCRIPTION
**Pull request by evantorrie**
Addresses #6123

I wasn't really sure where to add a test for this, and I don't particularly like the way I've done so.  It also doesn't seem to be easy to create a javaagent version of this (I think it would involve creating an Instrumentation extension), so I've restricted the test to just the "library" version of the instrumentation.

Feel free to completely change the test structure.

### CODE REVIEW

**Comment by trask** [comment]
thx @evantorrie! I sent a PR to your PR with some suggestions to align a bit more with existing tests (https://github.com/evantorrie/opentelemetry-java-instrumentation/pull/6)

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java [hunk 1]
```diff
@@ -5,31 +5,148 @@
...
+  private static final AttributeKey<String> CUSTOM_KEY = AttributeKey.stringKey("customKey");
+
+  private static final Metadata.Key<String> CUSTOM_METADATA_KEY =
+      Metadata.Key.of("customMetadataKey", Metadata.ASCII_STRING_MARSHALLER);
+
   @Override
   protected ServerBuilder<?> configureServer(ServerBuilder<?> server) {
     return server.intercept(
-        GrpcTelemetry.create(testing.getOpenTelemetry()).newServerInterceptor());
+        GrpcTelemetry.builder(testing.getOpenTelemetry()).build().newServerInterceptor());
```
Comment (trask) [inline instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java:45]: my bad
```suggestion
        GrpcTelemetry.create(testing.getOpenTelemetry()).newServerInterceptor());
```

File: instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java [hunk 2]
```diff
@@ -5,31 +5,148 @@
...
   }
 
   @Override
   protected ManagedChannelBuilder<?> configureClient(ManagedChannelBuilder<?> client) {
     return client.intercept(
-        GrpcTelemetry.create(testing.getOpenTelemetry()).newClientInterceptor());
+        GrpcTelemetry.builder(testing.getOpenTelemetry()).build().newClientInterceptor());
```
Comment (trask) [inline instrumentation/grpc-1.6/library/src/test/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTest.java:51]:
```suggestion
        GrpcTelemetry.create(testing.getOpenTelemetry()).newClientInterceptor());
```

**Review by trask — APPROVED** [review]
thx @evantorrie!

**Review by laurit — APPROVED** [review]

**Pull request merged**